services/code.service.js

```diff
@@ -119,61 +118,65 @@ const _prepareErrorMessage = (outputLog, language, command) => {
+                    {
+                        role: 'user',
+                        content: prompt,
+                    },
+                ],
+                model: langConfig.model,
+                response_format: {
+                    type: 'json_object',
                 },
-            ],
-            model: langConfig.model,
-            response_format: {
-                type: 'json_object',
-            },
-        })
-        let openAIResponse = {}
-        if (completion && completion.choices && completion.choices[0] && completion.choices[0].message) {
-            openAIResponse = JSON.parse(completion.choices[0].message.content)
-        }
-        let schema
-        if (systemPrompt) {
-            schema = Joi.any()
-        } else {
-            schema = Joi.object({
+                temperature: 0.1,
+            }),
+        )
+
+        const evaluatedResponses = (await Promise.all(promises)).map((res) => {
+            let openAIResponse = {}
+            if (res.choices[0]?.message) {
+                openAIResponse = JSON.parse(res.choices[0].message.content)
+            }
+
+            const schema = Joi.object({
                 score: Joi.number().integer().required(),
                 rationale: Joi.object({
                     positives: Joi.string().required().allow(''),
                     negatives: Joi.string().required().allow(''),
                 }).required(),
                 points: Joi.number().integer().required(),
             })
-        }
-        const validatedData = schema.validate(openAIResponse)
-        if (validatedData.error) {
-            /**
-             * 502 Bad Gateway
-             * This is often used when a server, while acting as a gateway or proxy, received an invalid response from the upstream server it accessed in attempting to fulfill the request.
-             */
-            logger.error(openAIResponse)
-            const responseCode = 502
-            const errorMessage = 'Unable to parse OPEN AI response'
-            response.errorMessage = errorMessage
-            response.statusCode = responseCode
-            response.error = 1
-        } else {
-            response.output = openAIResponse
-        }
+
+            const validatedData = schema.validate(openAIResponse)
+            if (validatedData.error) {
+                /**
+                    * 502 Bad Gateway
+                    * This is often used when a server, while acting as a gateway or proxy, received an invalid response from the upstream server it accessed in attempting to fulfill the request.
+                */
+                logger.error(openAIResponse)
+                const responseCode = 502
+                const errorMessage = 'Unable to parse OPEN AI response'
+                response.errorMessage = errorMessage
+                response.statusCode = responseCode
+                response.error = 1
+            } else {
+                return openAIResponse
+            }
```

Comment:
If I am not wrong the case where we have `validatedData.error` as set we are just setting couple of values for `response` and not returning anything as we are doing in else block. Is this expected ? Because we may end up returning `undefined` for that particular promise and that needs to be handled accordingly in `_calculateScoreConfidence`. Please check this once.